x/auth/types/credentials.go

```diff
@@ -32,25 +33,25 @@ const ModuleCredentialType = "ModuleCredential"
 
 var _ cryptotypes.PubKey = &ModuleCredential{}
 
-func NewModuleCredential(moduleName string, derivationKeys [][]byte) *ModuleCredential {
+// NewModuleCredential creates new module credential key.
+// At least one derivation key must be provided and all derivation keys must be not empty.
+func NewModuleCredential(moduleName string, derivationKeys ...[]byte) (*ModuleCredential, error) {
+	if len(derivationKeys) == 0 {
+		return nil, errors.New("module credential requires non empty derlivation key slice")
```

Comment:
I thought we already agreed that derivation keys are optional and that we can have a root module account which is what ADR 033 has always specified since the beginning.

Reply:
Good catch. Updating to make this case a non-error.